src/gui/qgsadvanceddigitizingfloater.cpp

```diff
@@ -0,0 +1,295 @@
+  connect( cadDockWidget, &QgsAdvancedDigitizingDockWidget::enabledChangedY, this, &QgsAdvancedDigitizingFloater::enabledChangedY );
+  connect( cadDockWidget, &QgsAdvancedDigitizingDockWidget::enabledChangedAngle, this, &QgsAdvancedDigitizingFloater::enabledChangedAngle );
+  connect( cadDockWidget, &QgsAdvancedDigitizingDockWidget::enabledChangedDistance, this, &QgsAdvancedDigitizingFloater::enabledChangedDistance );
+
+  // Connect our line edits signals to update cadDockWidget's tate
+  connect( mXLineEdit, &QLineEdit::returnPressed, cadDockWidget, [ = ]() { cadDockWidget->setX( mXLineEdit->text() ); } );
+  connect( mYLineEdit, &QLineEdit::returnPressed, cadDockWidget, [ = ]() { cadDockWidget->setY( mYLineEdit->text() ); } );
+  connect( mAngleLineEdit, &QLineEdit::returnPressed, cadDockWidget, [ = ]() { cadDockWidget->setAngle( mAngleLineEdit->text() ); } );
+  connect( mDistanceLineEdit, &QLineEdit::returnPressed, cadDockWidget, [ = ]() { cadDockWidget->setDistance( mDistanceLineEdit->text() ); } );
+}
+
+bool QgsAdvancedDigitizingFloater::eventFilter( QObject *obj, QEvent *event )
+{
+  if ( mCadDockWidget->cadEnabled() && mActive )
+  {
+    if ( event->type() == QEvent::MouseMove )
+    {
+      // We update the position when mouse moves
+      QMouseEvent *mouseEvent = dynamic_cast<QMouseEvent *>( event );
+      updatePos( mouseEvent->pos() );
+    }
+    else if ( event->type() == QEvent::Enter )
+    {
+      // We show the widget when mouse enters
+      QEnterEvent  *enterEvent = dynamic_cast<QEnterEvent *>( event );
+      updatePos( enterEvent->pos() );
+      setVisible( true );
+    }
+    else if ( event->type() == QEvent::Leave )
+    {
+      // We hide the widget when mouse leaves
+      setVisible( false );
+    }
+  }
+  return QWidget::eventFilter( obj, event );
+}
+
+bool QgsAdvancedDigitizingFloater::active()
+{
+  return mActive;
+}
+
+void QgsAdvancedDigitizingFloater::setActive( bool active )
+{
+  mActive = active;
+  if ( !active )
+  {
+    setVisible( false );
+  }
+}
+
+void QgsAdvancedDigitizingFloater::updatePos( QPoint pos )
+{
+  // We hardcode a small delta between the mouse position and the widget's position
+  move( pos + QPoint( 15, 5 ) );
+}
+
+void QgsAdvancedDigitizingFloater::changeX( QString text )
+{
+  mXLineEdit->setText( text );
+}
+
+void QgsAdvancedDigitizingFloater::changeY( QString text )
+{
+  mYLineEdit->setText( text );
+}
+
+void QgsAdvancedDigitizingFloater::changeDistance( QString text )
+{
+  mDistanceLineEdit->setText( text );
+}
+
+void QgsAdvancedDigitizingFloater::changeAngle( QString text )
+{
+  mAngleLineEdit->setText( text );
+}
+
+void QgsAdvancedDigitizingFloater::changeLockX( bool locked )
+{
+  if ( !locked )
+  {
+    mXLineEdit->setStyleSheet( "" );
+    mXLabel->setStyleSheet( "" );
+  }
+  else
+  {
+    mXLineEdit->setStyleSheet( "font-weight: bold" );
```

Comment:
```suggestion
    mXLineEdit->setStyleSheet( QStringLiteral( "font-weight: bold" ) );
```

Also applies to the others below.

Reply:
Ok